### ISSUE TITLE
Cannot export a notebook in html for tomboy-ng tools

### ISSUE THREAD

**Issue by gottogitgoing** [tomboy-notes/tomboy-ng#249]
The title pretty much says it all. The ui has options to export notebooks in HTML but when you attempt to do so the following error appears "Error, out format not set." This feature is essential for me because the notes I take in this notebook are used on an entire website. Additionally there is no distinction between the title for both tomboy-ng and tomboy-ng tools in the mate menu on linux. They both simply read tomboy-ng and because the icons for these entries are small you have no idea what will come up.
![Screenshot at 2021-08-28 18-50-51](https://user-images.githubusercontent.com/48987327/131212292-f85c2b6c-81d7-4ff5-be0b-89ffed16c38d.png)

**Comment by davidbannon** [comment]
Welcome to the bleeding edge gottogitgoing !
I just managed to replicate your problem, that good,  seems to happen when you select a NoteBook as source.  I will see if I can work out why and push up a fix.

And the menu problem sounds like just plain carelessness on my part, I used the tomboy-ng build scripts as a template, sounds like I missed changing at least that one. I'll fix it too. 

Sorry about those things, TT is not yet a very well tested app, probably will never be. But your feedback helps.
 
Davo

**Comment by davidbannon** [comment]
OK, both issues fixed in ver 0.6a, I have pushed up only Linux packages so far, looks like you are using Linux anyway ?  I will do a Windows one as soon as time permits.

Now, gottogitgoing, I am not sure that the install process will replace that stupid duplicated 'tomboy-ng', if not you will have to remove it manually with the MainMenu editor. If you need help with that, give me a yell, I use Mate too !

Davo

**Comment by gottogitgoing** [comment]
Davo
Thanks for the work I'll wait for the changes to come through in the updates. I posted this issue understanding the application was pretty fresh :))). In terms of the icons I've done as you've suggested and simply re-named them. Thanks again for the work.

Peter

**Comment by davidbannon** [comment]
>I'll wait for the changes to come through in the updates. 

Sorry, I do not understanding you there. The changes (for Linux) are up in the repository now. Or are you waiting for a Windows version ?   If so, some time tomorrow, its just on midnight here now ...

Davo

**Comment by gottogitgoing** [comment]
No, sorry I meant I'll check it out when I've applied the latest updates which I've just done and sadly there appears to be no change.

**Comment by davidbannon** [comment]
OK, thats pretty weird.  So I just installed that new version on my system (I don't usually, just have the binary in my ~/bin) and checked with github. I note some things -
* Firstly - v0.06a fixes both the problem you mention.
* Secondly, I still have a build script issue, the new binary does not show the new version number, its still 0.06. Sigh ...
* Thirdly, github tells me that there have not been any downloads of v0.6a. So, I wonder if you are getting some caching effect ?

I have removed the initial v0.06a ones and replaced them with ones with the same version number (yeah, I know, should not do that) that do display the correct version number. Can you please try again, make sure you download a package that has 0.06a as part of its file name. And, perhaps, to be sure, remove the old one before installing the new one ?

Again, I am only talking about Linux versions here, if you are looking for a Windows or Mac fix, we are going about it all wrong !

Davo

**Comment by gottogitgoing** [comment]
Davo
I've tried this again and no luck. The version I have on the ui has remained unchanged at v0.06 so there is definitely something weird going on. The output from apt I'll include in the screenshot below. 
![Tomboy-ng update 2021-08-29 17-38-44](https://user-images.githubusercontent.com/48987327/131242903-a2a35d8b-1d2c-4175-aabd-7c71b3fa7a0b.png)

**Comment by davidbannon** [comment]
OK, we have a problem here.  From the screen dump you just sent, it appears you are trying to update tomboy-ng from a distro repository. 

You need to update TomboyTools which is not in any repository, you need to download it yourself from github. https://github.com/davidbannon/TomboyTools/releases

tomboy-ng and TT are completely different packages. I have put tomboy-ng into both Debian and an Ubuntu PPA but as its a fair bit of work, I have not bothered to do so with TT. 

So, please download the appropriate package from https://github.com/davidbannon/TomboyTools/releases   Then, where you can see the downloaded file, 

 ` sudo apt install ./tomboytools_0.06a-0_amd64.deb [enter]`

That command line is based on my guess that you are using a 64bit Ubuntu (or ubuntu like) system. Will vary a bit ....

Davo

**Comment by gottogitgoing** [comment]
Davo
All good, had a little trouble finding the .deb package but after expanding "Assets" at the bottom of the page I found it. After trying it out I noticed the formatting includes a lot of paragraphs for line breaks. 

Comparing the html page source with the old Tomboy they used a div and css to style the whole page which negated the need for paragraph tags. Generally this has a better effect with an underlined title that spans the page, better spacing, and the font family for the body set to FreeSans.

I'll include the output source below in a tar file for comparison. I'm sure making these small changes wouldn't be difficult and vastly improve the output for users.

Peter
[html_source.tar.gz](https://github.com/tomboy-notes/tomboy-ng/files/7073201/html_source.tar.gz)

**Comment by davidbannon** [comment]
Yes, I am not sure why "Assets" needs to be expanded in TomboyTools but not tomboy-ng. I will have to have a look through the settings.

I have avoided using css in the HTML export to make it easy for someone to define and add their own CSS. I guess it depends a lot on what people want to do with it. Maybe I could -

* Add a tickbox to do plain or pre determined CSS ?
* Allow the user to add their own CSS definitions before its generated ?

I will look at the sample you have provided, thanks, will see just what can be done.  I have to warn you that TT is not a high priority for me at present, as I said in the release notes, TomboyTools is a tool I made for my use and posted on Github to keep the code safe.

But will look at it when I get some free time. 

Davo

**Comment by davidbannon** [comment]
Hi Peter, sorry I have taken so long but I have had a look at the CSS you sent me.  I have just pushed out a new release based on it, here -

https://github.com/davidbannon/TomboyTools/releases/tag/v0.07

Now, I say 'based' because I have not used the exact Tomboy CSS.  I did not like several "look" issues with the original, particularly relating to Lists and to how the unrendered HTML looks. So, I have made a few minor changes. 

If you depend on the output looking exactly the same as the Tomboy original, then please let me know and I will find a solution. Otherwise, have a look at what I have done if you can please.

Davo
